src/main/docker/Dockerfile

```diff
@@ -66,6 +66,13 @@ RUN chgrp 0 /usr/share/opensearch/opensearch-docker-entrypoint.sh && \
 COPY scripts/ /opt/opensearch/scripts/
 RUN chmod +x /opt/opensearch/scripts/*
 
+RUN zypper -n remove which hostname netcat-openbsd unzip wget
+
+#
+# The actual image definition
+#
+FROM builder as opensuse-opensearch1
```

Comment:
The way you're using the builder image here it isn't actually achieving anything.

If you take a look at the image produced with `docker history`, which outputs a line for each layer, you can see that it will has a layers where all the tools are added and then extra layers for removing them again.

**Command**

    docker image history --no-trunc --format '{{.CreatedBy}}' cafapi/prereleases:opensuse-opensearch1-2.8.0-US898097-SNAPSHOT | tac

<details>
<summary><b>Results</b></summary>

    KIWI 9.24.43
    /bin/sh -c #(nop)  ENV LANG=en_US.utf8
    /bin/sh -c #(nop) COPY file:c21b7051a55cc1db3a76ae38ecef52f79c9c992c7d79867bb669a416c7962f5c in /etc/crypto-policies/policies/modules 
    |0 /bin/sh -c zypper -n refresh &&     zypper -n update &&     zypper -n install curl postgresql dejavu-fonts crypto-policies-scripts &&     update-crypto-policies --set DEFAULT:DISABLE-CBC &&     sed -rie '/^jdk.tls.disabledAlgorithms=/s/$/, TLS_DHE_RSA_WITH_CHACHA20_POLY1305_SHA256/'         /etc/crypto-policies/back-ends/java.config &&     zypper -n remove -u crypto-policies-scripts &&     zypper -n clean --all
    /bin/sh -c #(nop) COPY file:d35dc053a03732cfc7a97e9be02d7804d353abc8ea5dd2221226d294c5f25bbe in /usr/local/bin/su-exec 
    /bin/sh -c #(nop) COPY dir:dc3d28500c6fc303aab11f67039854cfab3737046f643bace5acf42820c79e9c in /startup 
    /bin/sh -c #(nop) ADD 28e4f74df7e180b0d614b5177b8f5027e7ba00cd9c913b8d75dc9d3f4a4a1ee0 in /startup/startup.d/install-ca-cert-base.sh 
    |0 /bin/sh -c chmod +x /startup/* /startup/startup.d/*
    |0 /bin/sh -c ln -s /usr/local/bin/su-exec /usr/local/bin/gosu
    /bin/sh -c #(nop)  ENV TINI_VERSION=v0.19.0
    /bin/sh -c #(nop) ADD 419422cadcf348cc19b91714baa6298d7d99ff5a0b42aab61b6ef29fe3abc403 in /tini 
    |0 /bin/sh -c chmod +x /tini
    /bin/sh -c #(nop)  ENTRYPOINT ["/tini" "--" "/startup/startup.sh"]
    /bin/sh -c #(nop) COPY dir:2495d4a448324301ff31a939dc039586c08da399484670ed432cd1b2d588b707 in /scripts 
    |0 /bin/sh -c chmod +x /scripts/*
    LABEL Build.Date=2024-03-08T12:29:16.145Z Build.Number=2.10.0-673 Git.Repo=https://github.com/CAFapi/opensuse-base-image Git.Branch=develop Git.Commit=75960c07ad34d94a6144455bb7c1a19eefec9b33
    |0 /bin/sh -c zypper -n refresh &&     zypper -n update &&     zypper -n install java-17-openjdk &&     zypper -n clean --all
    /bin/sh -c #(nop) ADD 6c3eaf02434384746e4003295e4b291bf1f48c6ae895c2e0cd3c1dc29b306929 in /startup/startup.d/ 
    |0 /bin/sh -c chmod +x /startup/startup.d/install-ca-cert-java.sh
    /bin/sh -c #(nop)  ENV JRE_HOME=/usr/lib64/jvm/java-17-openjdk-17
    /bin/sh -c #(nop) COPY dir:a846d97616e7d513467be9525876868d24c9308cb44b24003f559067c0da1a7d in /etc/pki/nssdb 
    LABEL Build.Date=2024-03-08T13:56:13.918Z Build.Number=1.5.0-261 Git.Repo=https://github.com/CAFapi/opensuse-java17-images Git.Branch=main Git.Commit=370a66b7aaa1c24b4443e8c519f509b8d0c2a258
    |0 /bin/sh -c zypper -n refresh &&     zypper -n update
    /bin/sh -c #(nop) COPY dir:4a10b1da987fb6a092594ef05589b2ef356b4a08a2484d314c7cc5de0252b112 in /etc/pki/ca-trust 
    |0 /bin/sh -c zypper -n install which hostname netcat-openbsd unzip wget
    |0 /bin/sh -c zypper -n clean --all
    |0 /bin/sh -c mkdir -p /usr/share/opensearch
    |0 /bin/sh -c groupadd -g 1000 opensearch &&     useradd -u 1000 -g 1000 -G 0 -d /usr/share/opensearch opensearch &&     chmod 0775 /usr/share/opensearch &&     chgrp 0 /usr/share/opensearch
    /bin/sh -c #(nop) WORKDIR /usr/share/opensearch
    /bin/sh -c #(nop) COPY --chown=1000:0dir:0cc22fc0abb1b716301490491c5a1f8e5382a12c3bca5b25f8885b3e96ee4b4d in /usr/share/opensearch/ 
    /bin/sh -c #(nop) COPY file:21f3369453fc9caa163c21e625f4714e1adcaa4985e1c6a09f4bb646f842bf3a in /usr/share/opensearch/config 
    /bin/sh -c #(nop) COPY multi:c570d262cdbc737f1231977b9eb7ca011e3f10cf5123ba1374dd762402cbabc1 in /usr/share/opensearch/lib/ 
    |0 /bin/sh -c JAVA_HOME=${JRE_HOME} OPENSEARCH_JAVA_OPTS="$(echo ${https_proxy} | sed -E 's/^.*:\/\/(.*):([0-9]*).*/-Dhttps.proxyHost=\1 -Dhttps.proxyPort=\2/')"     /usr/share/opensearch/bin/opensearch-plugin install analysis-icu
    /bin/sh -c #(nop)  ENV PATH=/usr/share/opensearch/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
    /bin/sh -c #(nop) COPY --chown=1000:0file:554c8135cafd81f3b6de411601cccce88918d70c30499fc89f003178a98458b1 in /usr/share/opensearch/opensearch-docker-entrypoint.sh 
    |0 /bin/sh -c chgrp 0 /usr/share/opensearch/opensearch-docker-entrypoint.sh &&     chmod g=u /etc/passwd &&     chmod 0775 /usr/share/opensearch/opensearch-docker-entrypoint.sh
    /bin/sh -c #(nop) COPY dir:56c8080279cc58d56daac8b948b7393e228891e6af777d63c0a2628e5b831d82 in /opt/opensearch/scripts/ 
    |0 /bin/sh -c chmod +x /opt/opensearch/scripts/*
    |0 /bin/sh -c zypper -n remove which hostname netcat-openbsd unzip wget
    /bin/sh -c #(nop)  EXPOSE 9200 9300
    /bin/sh -c #(nop)  HEALTHCHECK &{["CMD-SHELL" "curl -k -u \"admin:admin\" https://0.0.0.0:9200 || exit 1"] "1m0s" "0s" "5m0s" '\x00'}
    /bin/sh -c #(nop)  CMD ["/bin/sh" "-c" "/opt/opensearch/scripts/start.sh"]
    LABEL Build.Date=2024-03-21T16:44:37.588Z Build.Number=2.8.0-US898097-SNAPSHOT Git.Repo=https://github.com/CAFapi/opensuse-opensearch1-image Git.Branch=US898097 Git.Commit=ada69421319b2b2d71159701356349fd7a4b7142

</details>

Reply:
I have reworked this and the results on the `docker history` look better to me:
<details>
<summary><b>Results</b></summary>

KIWI 9.24.43
/bin/sh -c #(nop)  ENV LANG=en_US.utf8
/bin/sh -c #(nop) COPY file:c21b7051a55cc1db3a76ae38ecef52f79c9c992c7d79867bb669a416c7962f5c in /etc/crypto-policies/policies/modules
|0 /bin/sh -c zypper -n refresh &&     zypper -n update &&     zypper -n install curl postgresql dejavu-fonts crypto-policies-scripts &&     update-crypto-policies --set DEFAULT:DISABLE-CBC &&     sed -rie '/^jdk.tls.disabledAlgorithms=/s/$/, TLS_DHE_RSA_WITH_CHACHA20_POLY1305_SHA256/'         /etc/crypto-policies/back-ends/java.config &&     zypper -n remove -u crypto-policies-scripts &&     zypper -n clean --all
/bin/sh -c #(nop) COPY file:d35dc053a03732cfc7a97e9be02d7804d353abc8ea5dd2221226d294c5f25bbe in /usr/local/bin/su-exec
/bin/sh -c #(nop) COPY dir:dc3d28500c6fc303aab11f67039854cfab3737046f643bace5acf42820c79e9c in /startup
/bin/sh -c #(nop) ADD 28e4f74df7e180b0d614b5177b8f5027e7ba00cd9c913b8d75dc9d3f4a4a1ee0 in /startup/startup.d/install-ca-cert-base.sh
|0 /bin/sh -c chmod +x /startup/* /startup/startup.d/*
|0 /bin/sh -c ln -s /usr/local/bin/su-exec /usr/local/bin/gosu
/bin/sh -c #(nop)  ENV TINI_VERSION=v0.19.0
/bin/sh -c #(nop) ADD 419422cadcf348cc19b91714baa6298d7d99ff5a0b42aab61b6ef29fe3abc403 in /tini
|0 /bin/sh -c chmod +x /tini
/bin/sh -c #(nop)  ENTRYPOINT ["/tini" "--" "/startup/startup.sh"]
/bin/sh -c #(nop) COPY dir:2495d4a448324301ff31a939dc039586c08da399484670ed432cd1b2d588b707 in /scripts
|0 /bin/sh -c chmod +x /scripts/*
LABEL Build.Date=2024-03-08T12:29:16.145Z Build.Number=2.10.0-673 Git.Repo=https://github.com/CAFapi/opensuse-base-image Git.Branch=develop Git.Commit=75960c07ad34d94a6144455bb7c1a19eefec9b33
|0 /bin/sh -c zypper -n refresh &&     zypper -n update &&     zypper -n install java-17-openjdk &&     zypper -n clean --all
/bin/sh -c #(nop) ADD 6c3eaf02434384746e4003295e4b291bf1f48c6ae895c2e0cd3c1dc29b306929 in /startup/startup.d/
|0 /bin/sh -c chmod +x /startup/startup.d/install-ca-cert-java.sh
/bin/sh -c #(nop)  ENV JRE_HOME=/usr/lib64/jvm/java-17-openjdk-17
/bin/sh -c #(nop) COPY dir:a846d97616e7d513467be9525876868d24c9308cb44b24003f559067c0da1a7d in /etc/pki/nssdb
LABEL Build.Date=2024-03-08T13:56:13.918Z Build.Number=1.5.0-261 Git.Repo=https://github.com/CAFapi/opensuse-java17-images Git.Branch=main Git.Commit=370a66b7aaa1c24b4443e8c519f509b8d0c2a258
|0 /bin/sh -c zypper -n refresh &&     zypper -n update
/bin/sh -c #(nop) COPY dir:878ad3b3ac6473678fe436c5e9a7598ea9d6994de0e9b1a2519d86b5df12043c in /etc/pki/ca-trust
|0 /bin/sh -c mkdir -p /usr/share/opensearch
|0 /bin/sh -c groupadd -g 1000 opensearch &&     useradd -u 1000 -g 1000 -G 0 -d /usr/share/opensearch opensearch &&     chmod 0775 /usr/share/opensearch &&     chgrp 0 /usr/share/opensearch
/bin/sh -c #(nop) COPY --chown=1000:0dir:80d72b3c1f40fab4bd2c6ba06d398e49335a43ca1d76e7f6f49e6b3f7f012cb0 in /usr/share/opensearch/
/bin/sh -c #(nop)  ENV PATH=/usr/share/opensearch/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
/bin/sh -c #(nop) COPY dir:bf798750ce64174b296e14ca1f0f27d8640701dc9b1abb89ebc114933968b551 in /opt/opensearch/scripts/
|0 /bin/sh -c chmod +x /opt/opensearch/scripts/*
/bin/sh -c #(nop)  EXPOSE 9200 9300
/bin/sh -c #(nop)  HEALTHCHECK &{["CMD-SHELL" "curl -k -u \"admin:admin\" https://0.0.0.0:9200 || exit 1"] "1m0s" "0s" "5m0s" '\x00'}
/bin/sh -c #(nop)  CMD ["/bin/sh" "-c" "/opt/opensearch/scripts/start.sh"]

</details>